### PR TITLE
feat: [DHIS2-13681] optional decimal places argument for d2:round

### PR DESCRIPTION
**Pull request by superskip**

### CODE REVIEW

File: src/core_modules/capture-core-utils/rulesEngine/d2Functions/getD2Functions.js [hunk 1]
```diff
@@ -21,8 +21,14 @@ export const getD2Functions = ({
         execute: (params: any) => Math.floor(params[0]),
     },
     round: {
-        parameters: 1,
-        execute: (params: any) => Math.round(params[0]),
+        parameters: { min: 1, max: 2 },
+        execute: (params: any) => {
+            if (params[1]) {
+                const decimalCorrection = 10 ** Math.round(params[1]);
+                return Math.round((params[0] + Number.EPSILON) * decimalCorrection) / decimalCorrection;
```
Comment (simonadomnisoru) [inline src/core_modules/capture-core-utils/rulesEngine/d2Functions/getD2Functions.js:28]: NIT: probably `return +Math.round(params[0]).toFixed(params[1])` also works

**Review by simonadomnisoru — APPROVED** [review]
LGTM!

**Review by jasminenguyennn — APPROVED** [review]

File: src/core_modules/capture-core-utils/rulesEngine/d2Functions/getD2Functions.js (hunk 1 above)
Comment (superskip) [reply src/core_modules/capture-core-utils/rulesEngine/d2Functions/getD2Functions.js:28]: Thanks for the review!
I think your solution is what we actually want here :+1:

Comment (superskip) [reply src/core_modules/capture-core-utils/rulesEngine/d2Functions/getD2Functions.js:28]: Oops, no, it doesn't work.. `params[1] == 1` means 9.28 should be rounded to 9.3, and that can't be done with `Math.round` unless one first scales the number.

**Review by geethaalwan — APPROVED** [review]
Tested successfully on 2.40 version

**Pull request merged**